### PR TITLE
Movie Screen layout created, collections view created, banner basic l…

### LINKED ISSUE #8: Feature - Movie Screen
Description:

As an user, I want to see the movies as cards with horizontal scroll, with the categories:
- Now Movies
- Popular Movies
- Incoming Movies
At the top of the view a title of section "Movies" and a banner with random movies with some details

Checklist:
Create custom xib collection item
Create collections view
Create a Model Repository for API request

**Epic:** #2

### PR DESCRIPTION
**Pull request by JGTobares**
- Movie Screen Layout
- Collections View for NowMovies, PopularMovies and UpcomingMovies
- Basic banner layout
- Navigation Bar with menu and search button
- Assets yellow star, menu and search icon added

### CODE REVIEW

File: MovieApp/MovieApp/Model/MovieManager.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//
+//  MovieManager.swift
+//  MovieApp
+//
+//  Created by Julio Gabriel Tobares on 14/04/2022.
+//
+
+import Foundation
+
+protocol MovieManagerDelegate {
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/Model/MovieManager.swift]: Move this to Protocols file in Supporting

File: MovieApp/MovieApp/Model/MovieManager.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    func onNowLoaded(movie: Result<[Movie], CustomError>)
+    func onPopularLoaded(movie: Result<[Movie], CustomError>)
+    func onUpcomingLoaded(movie: Result<[Movie], CustomError>)
+}
+
+class MovieManager {
+    var nowMovies:[Movie] = []
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/Model/MovieManager.swift]: These lines need indentation (space after ':')

File: MovieApp/MovieApp/ViewController/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+//  MovieApp
+//
+//  Created by Julio Gabriel Tobares on 12/04/2022.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet var NowMovies: UICollectionView!
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/ViewController/ViewController.swift]: Use camel case for variables including outlets

File: MovieApp/MovieApp/ViewController/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    func refreshMovies() {
+        DispatchQueue.main.async { [weak self] in
+            self?.NowMovies.reloadData()
+            self?.PopularMovies.reloadData()
+            self?.UpcomingMovies.reloadData()
+        }
+    }
+    /*
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/ViewController/ViewController.swift]: Delete comment

File: MovieApp/MovieApp/ViewController/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+extension ViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        NowMovies.deselectItem(at: indexPath, animated: true)
+        
+        print("Item pressed")
+    }
+}
+
+extension ViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/ViewController/ViewController.swift:66]: Add empty line after extension line and remove empty line after functions lines

File: MovieApp/MovieApp/Model/MovieManager.swift [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+//  MovieApp
+//
+//  Created by Julio Gabriel Tobares on 14/04/2022.
+//
+
+import Foundation
+
+protocol MovieManagerDelegate {
+    func onNowLoaded(movie: Result<[Movie], CustomError>)
+    func onPopularLoaded(movie: Result<[Movie], CustomError>)
```
Comment (Fernando-GM) [inline MovieApp/MovieApp/Model/MovieManager.swift]: Don't use 'Result<>' with delegate. You are already handling Result cases in Manager. If, and only if, you use the data sent in delegate, use '[Movie]' instead and change parameter to 'movies'. If you don't use data sent in delegate, don't have parameters

**Review by Fernando-GM — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/Model/MovieManager.swift (hunk 1 above)
Comment (JGTobares) [reply MovieApp/MovieApp/Model/MovieManager.swift]: Protocol moved to Supporting

File: MovieApp/MovieApp/Model/MovieManager.swift (hunk 2 above)
Comment (JGTobares) [reply MovieApp/MovieApp/Model/MovieManager.swift]: Done!

File: MovieApp/MovieApp/ViewController/ViewController.swift (hunk 3 above)
Comment (JGTobares) [reply MovieApp/MovieApp/ViewController/ViewController.swift]: Names fixed

File: MovieApp/MovieApp/ViewController/ViewController.swift (hunk 4 above)
Comment (JGTobares) [reply MovieApp/MovieApp/ViewController/ViewController.swift]: Comment deleted

File: MovieApp/MovieApp/Model/MovieManager.swift (hunk 6 above)
Comment (JGTobares) [reply MovieApp/MovieApp/Model/MovieManager.swift]: Argument of each func deleted

**Review by Fernando-GM — APPROVED** [review]

**Pull request merged**